### ISSUE TITLE
[BatchNorm1d] Question on the gradient of bias

### ISSUE THREAD

**Issue by CDLuminate** [pytorch/pytorch#2330]
According to https://kratzert.github.io/2016/02/12/understanding-the-gradient-flow-through-the-batch-normalization-layer.html
, the gradient of the bias in a BatchNorm1d layer should be `gradOutput.sum(0)` given that the size of gradOutput is `(N,D)`.

```
#step9
dbeta = np.sum(dout, axis=0)
```

However, when I'm verifying this, pytorch seems to use `input.sum(0)` as the bias gradient.

Well ... if we formulate the affine part as `y = vx + b`, then `∂y/∂b = x` ??

I searched the code and the only part that checked BatchNorm1d layer just uses one without affine transformation. `test_autograd.py`
```python
 307     def test_hooks_cpp(self):                                                   
 308         # Tests hooks for autograd function implemented in C++                  
 309         bn = torch.nn.BatchNorm1d(5, affine=False)                              
 310         bn.eval()                                                               
 311                                                                                 
 312         counter = [0]                                                           
 313                                                                                 
 314         def bw_hook(grad):                                                      
 315             counter[0] += 1                                                     
 316             return grad * 2                                                     
 317                                                                                 
 318         x = Variable(torch.ones(5, 5), requires_grad=True)                      
 319         z = bn(x)                                                               
 320         z.register_hook(bw_hook)                                                
 321         z.sum().backward()                                                      
 322                                                                                 
 323         self.assertEqual(counter[0], 1, 'bw_hook not called')                   
 324         self.assertEqual(x.grad.data, torch.ones(5, 5) * 2)                     
```

Code to reproduce:
```python
import torch as th
from torch.autograd import Variable
x = Variable(th.rand(3,5), requires_grad=True)
l = th.nn.BatchNorm1d(5)
y = l(x)

gy = Variable(th.ones(3,5), requires_grad=False)
l.zero_grad()
y.backward(x)

l.bias.grad
'''
Variable containing:
 1.6464
 1.4370
 1.6897
 1.9258
 1.6882
[torch.FloatTensor of size 5]
'''

# -- the expected gradient of bias
gy.sum(0)
'''
Variable containing:
 3
 3
 3
 3
 3
[torch.FloatTensor of size 5]
'''

x.sum(0)
'''
Variable containing:
 1.6464
 1.4370
 1.6897
 1.9258
 1.6882
[torch.FloatTensor of size 5]
'''
```

Is there something wrong?

```
In [31]: th.__version__
Out[31]: '0.2.0_1'
```

**Comment by CDLuminate** [comment]
Sorry, I've got a typo in this line `y.backward(x)`, it should be `y.backward(gy)` ...
